src/sagas/index.js

```diff
@@ -203,35 +203,32 @@ export function* fetchSingleVm (action) {
 
   action.payload.additional = shallowFetch ? VM_FETCH_ADDITIONAL_SHALLOW : VM_FETCH_ADDITIONAL_DEEP
   const vm = yield callExternalAction(Api.getVm, action, true)
-  const error = vm.error ? vm.error.status ?? 'fetch-error' : null
+  if (vm?.error) {
+    return { vmId, error: vm.error }
+  }
 
-  let internalVm = null
-  if (vm?.id) {
-    internalVm = yield transformAndPermitVm(vm)
+  const internalVm = yield transformAndPermitVm(vm)
 
-    // If the VM is running, we want to display the current=true cdrom info. Due
-    // to an API restriction, current=true cdrom info cannot currently (Aug-2018)
-    // be accessed via the additional fetch list on the VM. Fetch it directly.
-    if (!shallowFetch && internalVm.status === 'up') {
-      internalVm.cdrom = yield fetchVmCdRom({ vmId: internalVm.id, current: true })
-    }
+  // If the VM is running, we want to display the current=true cdrom info. Due
+  // to an API restriction, current=true cdrom info cannot currently (Aug-2018)
+  // be accessed via the additional fetch list on the VM. Fetch it directly.
+  if (!shallowFetch && internalVm.status === 'up') {
+    internalVm.cdrom = yield fetchVmCdRom({ vmId: internalVm.id, current: true })
+  }
 
-    if (!shallowFetch) {
-      yield parallelFetchAndPopulateSnapshotDisksAndNics(internalVm.id, internalVm.snapshots)
-    }
+  if (!shallowFetch) {
+    yield parallelFetchAndPopulateSnapshotDisksAndNics(internalVm.id, internalVm.snapshots)
   }
 
-  return { vmId, internalVm, error }
+  return { vmId, internalVm }
 }
 
 export function* fetchAndPutSingleVm (action) {
   const { internalVm, error } = yield fetchSingleVm(action)
 
-  if (error) {
-    if (error === 404) {
-      yield put(updateVms({ removeVmIds: [action.payload.vmId] }))
-    }
-  } else {
+  if (error?.status === 404) {
+    yield put(updateVms({ removeVmIds: [action.payload.vmId] }))
+  } else if (!error && internalVm) {
```

Comment:
I prefer the old if structure to the new one ... `if (error) { do errors } else { do success }`.